src/v/config/configuration.cc

```diff
@@ -110,6 +110,14 @@ configuration::configuration()
        .example = "31536000000",
        .visibility = visibility::tunable},
       24h * 365)
+  , log_storage_max_size(
```

Comment:
Maybe `target` rather than `max`: we are not systematically staying below the threshold, but rather will use approximately this much storage.